packages/types/src/index.ts

```diff
@@ -37,8 +35,10 @@ export interface RequestArguments {
 }
 
 // per EIP-1193
-export interface Provider extends EventEmitter {
+export interface Provider {
   request(args: RequestArguments): Promise<unknown>
+  on(event: any, listener: any): void
```

Comment:
This is a regression in the interface, because (1) you have fewer methods and (2) the return values have changed; and neither interface actually adheres to the spec. It's correct to only have `'on' | 'removeListener'`, but ["These methods MUST be implemented per the Node.js [EventEmitter API](https://nodejs.org/api/events.html)."](https://eips.ethereum.org/EIPS/eip-1193#events)

Would you change it to `export interface Provider extends Pick<EventEmitter, 'on' | 'removeListener'> {`? (I'm not sure if you can `Pick` from a class interface, but if you can I think that's the stablest way to express this.)

Reply:
Oh, good call. I was probably overly optimistic deriving that type information from WalletConnect connector, assuming it was EIP compliant. Will take another look and double check.

Reply:
I am going to ship an update that will fail until WC ships an update to their type definitions. 

> This is a regression in the interface, because (1) you have fewer methods 

Yes, this is a breaking change, but at the same time, it's a good breaking change, because it can't be guaranteed that all providers extend EventEmitter and it leads us to things such as `MockWalletConnectProvider` where we do type casting (confusing end developers that on/off/removeListener are spec-compliant, where they're not). It is also not guaranteed all providers extend from EventEmitter, since it's just a recommendation.

That said, I think it's a good change to ship and will make it easier to work with web3-react going forward.